### PR TITLE
osmosis 0.45

### PR DESCRIPTION
**Pull request by fabrik42**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

---

The audit gives a warning that no tests are defined. What's the best way to add a test for this rather simple formula?

Best,

Christian

### CODE REVIEW

**Comment by tdsmith** [comment]
Feed it some sample data and make sure it doesn't choke, ideally! You can download a small dataset with a `resource` if there isn't one in the tarball we're downloading.

**Comment by fabrik42** [comment]
Hmm sorry, but I am still unsure how to do this. I still have a couple of questions left:

Where is the dataset supposed to come from? Should I host it myself? And how can I prove that this works? You basically want to do a very simple operation to prove that it is callable?

Thanks,
Christian

**Comment by dunn** [comment]
> Where is the dataset supposed to come from? 

Anywhere reliable, where it can be expected to not disappear.

> You basically want to do a very simple operation to prove that it is callable?

Yep!

**Comment by fabrik42** [comment]
Ok, I just added the test. Instead of downloading something, I just provided a very simple fixture in the spec. I hope this is ok.

Thanks for the help!

File: Formula/osmosis.rb [hunk 1]
```diff
@@ -13,6 +13,32 @@ def install
     bin.install "bin/osmosis"
     libexec.install %w[lib config script]
   end
+
+  test do
+    path = testpath/"test.osm"
+    path.write <<-EOS
```
Comment (MikeMcQuaid) [inline Formula/osmosis.rb]: Use EOS.undent` and intent the below lines to be two-spaces beyond the whitespace on this line.

File: Formula/osmosis.rb [hunk 2]
```diff
@@ -13,6 +13,32 @@ def install
...
+  <tag k="shelter" v="yes"/>
+  <tag k="tactile_paving" v="no"/>
+  <tag k="wheelchair" v="no"/>
+  <tag k="wheelchair:description" v="Kein Kasseler Bord"/>
+</node>
+</osm>
+    EOS
+
+    `#{bin}/osmosis --read-xml file="#{path}" --write-null`
+    assert_equal 0, $?.exitstatus
```
Comment (MikeMcQuaid) [inline Formula/osmosis.rb]: Use `system` instead which will check the exit status.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/osmosis.rb (hunk 1 above)
Comment (fabrik42) [inline Formula/osmosis.rb]: done!

File: Formula/osmosis.rb (hunk 2 above)
Comment (fabrik42) [inline Formula/osmosis.rb]: good point!

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by fabrik42** [comment]
Thanks, you're welcome! :)

**Pull request merged**